### ISSUE TITLE
Error: Username with "_" (underscore) can't login.

### ISSUE THREAD

**Issue by compiler0** [guardianproject/ChatSecureAndroid#645]
I can't login with an username like _forexample_@jabber.org.

I've tried to login with different other clients(xabber,pidgin) and there are no login errors.

Thanks a lot!

**Comment by DjSlash** [comment]
It's caused by [this merge](https://github.com/guardianproject/ChatSecureAndroid/pull/621). Currently only alphanumeric usernames will be able to log in to their accounts.

@n8fr8 can you please fix this?

**Comment by n8fr8** [comment]
Yes, will do. Our regular expression got overzealous. Apologies.

**Comment by DjSlash** [comment]
@n8fr8 The RFC expresses that usernames may consist of more kinds of characters that isn't covered by your regexp. I wonder why you don't use nodeprep ? It will save you the trouble to update the regexp everytime someone uses something that isn't covered by it.

**Comment by n8fr8** [comment]
We should use nodeprep. That said, I hope our RC5 release solved your problem.

**Comment by DjSlash** [comment]
Yep, works for me.
